pkg/apis/management.cattle.io/v3/machine_types.go

```diff
@@ -269,28 +269,46 @@ type NodeCommonParams struct {
 }
 
 // +genclient
-// +kubebuilder:skipversion
 // +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Activated",type=string,JSONPath=".spec.active"
+// +kubebuilder:printcolumn:name="Binary URL",type=string,JSONPath=".spec.url"
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+// NodeDriver represents a Rancher node driver for a specific cloud provider used to provision cluster nodes.
 type NodeDriver struct {
 	metav1.TypeMeta `json:",inline"`
 	// Standard object’s metadata. More info:
 	// https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#metadata
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-	// Specification of the desired behavior of the the cluster. More info:
+	// Specification of the desired behavior of the cluster. More info:
 	// https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#spec-and-status
 	Spec NodeDriverSpec `json:"spec"`
 	// Most recent observed status of the cluster. More info:
 	// https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#spec-and-status
+	// +optional
 	Status NodeDriverStatus `json:"status"`
 }
 
 type NodeDriverStatus struct {
-	Conditions                  []Condition `json:"conditions"`
-	AppliedURL                  string      `json:"appliedURL"`
-	AppliedChecksum             string      `json:"appliedChecksum"`
-	AppliedDockerMachineVersion string      `json:"appliedDockerMachineVersion"`
+	// Conditions is a representation of the current state of the driver
+	// +optional
+	Conditions []Condition `json:"conditions"`
```

Comment:
Can we use the `metav1.Condition` here instead? Also with the following kubebuilder comments:
```
	// +optional
	// +listType=map
	// +listMapKey=type
	// +kubebuilder:validation:MaxItems=32
```
Maybe with a short description of which conditions are expected for the driver.

Reply:
I've moved the custom `Condition` struct into its own file and added a short blurb about this. I've also copied over the relevant upstream comments and kubebuilder markers.

lmk if we should remove the requirements markers, I think they make sense for what we expect for a node driver but I'm not 100% sure if the goal is purely documentation or also adding validation.